src/dstack/_internal/core/models/runs.py

```diff
@@ -416,7 +421,7 @@ def _error(cls, values) -> Dict:
 
 class JobPlan(CoreModel):
     job_spec: JobSpec
-    offers: List[InstanceOfferWithAvailability]
+    offers: List[Union[InstanceSharedOffer, InstanceOfferWithAvailability]]
```

Comment:
Do we really need to introduce different types of offers to the API? Can't every InstanceOfferWithAvailability be InstanceSharedOffer with total_blocks/blocks = 1?

Reply:
https://github.com/dstackai/dstack/pull/2253/commits/3342a34f1c371d3ff58ac5e41118e56c28a63059